packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts

```diff
@@ -49,6 +49,205 @@ test('has many query', () => {
+      transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError(); // TODO: Fix validation error messaging - 'Unknown type "Foo". Did you mean "Member"?'
+});
+
+test('fails if hasMany related type is not an array', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: Member @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: String
+      team: Team @belongsTo(references: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError(); // TODO: Error message
+});
+
+test('fails if uni-directional hasMany', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: [Member] @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: String
+      team: Team
+    }`;
+
+  // TODO: This should fail -- find appropriate place to validate.
+  // expect(() =>
+  //   testTransform({
+  //     schema: inputSchema,
+  //     transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+  //   }),
+  // ).toThrowError();
+});
+
+test('fails if used as a has one relationship', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: Member @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: String
+      team: Team @belongsTo(fields: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError('@hasMany must be used with a list. Use @hasOne for non-list types.');
+});
+
+test('hasMany / hasOne - belongsTo across data source type boundary', () => {
+  const mockSqlStrategy = mockSqlDataSourceStrategy();
+
+  const inputDdbSchema = `
+    type Member @model {
+      name: String
+      team: Team @belongsTo(references: "teamId")
+      teamId: String
+    }
+
+    type Project @model {
+      name: String
+      teamId: String
+      team: Team @belongsTo(references: "teamId")
+    }
+
+    type Team @model {
+      id: String! @primaryKey
+      mantra: String
+      members: [Member!] @hasMany(references: "teamId")
+      project: Project @hasOne(references: "teamId")
+    }
+  `;
+
+  const out = testTransform({
+    schema: inputDdbSchema,
+    transformers: [
+      new ModelTransformer(),
+      new PrimaryKeyTransformer(),
+      new HasOneTransformer(),
+      new HasManyTransformer(),
+      new BelongsToTransformer(),
+    ],
+    dataSourceStrategies: {
+      Member: DDB_DEFAULT_DATASOURCE_STRATEGY,
+      Project: DDB_DEFAULT_DATASOURCE_STRATEGY,
```

Comment:
It is better to have the amplify table strategy as well in this test case as it is a common case in Gen2